### ISSUE TITLE
Canary category

### ISSUE THREAD

**Issue by jamierocks** [BtoBastian/bStats#42]
CanaryMod is admittedly dead is some people's eyes - I prefer hibernating - we've used mcstats for some time now, internally to track users of CanaryMod and by extension NeptuneMod.

bStats has now been around long enough, that I'm happy to start using it with NeptuneMod and work its way into CanaryLib as a replacement for mcstats.

---

Talking more broadly than adding a Canary category, I'd like to pose the question as to whether having different categories is such a good idea? To me it seems like unnecessary fragmentation - forcing plugins to not be able to see their total stats from one location. As far as plugins go, platform should be irregardless when considering statistics **other than platform statistics of course**.

**Comment by BtoBastian** [comment]
Hey,
do you want to use bStats to track stats of CanaryMod itself, or do you want plugin owners who use Canary to be able to register plugins on bStats? If I understood you correctly you want both, right?
- Tracking stats for CanaryMod itself would be no big deal and I can setup everything's necessary in within a few minutes
- Allowing users to add CanaryMod plugins is a little bit complicated, as it would require an extra Metrics class which might take some time as I'm not familiar with the Canary API.

---

Different categories exist for serveral reason as already discussed in other issues which wanted an universal plugin category, so I won't go into details and just sum up some of the points:
- Incorrect data, e.g. player charts if a plugin supports both Bungeecord and Bukkit -> Players might be counted twice (which is happening in McStats afaik, even though I'm not 100% sure)
- Even though you can put all code for the different implementations in one universal jar file, they are technically sesperate implementations
- There can be different charts for different implementations (e.g. Bungeecord having an extra "Proxy server" line charts)
- It's way more future proof, as I want to support pocket edition servers and user clients in the future, too
- Users of different implementations can have different habits, e.g. which Minecraft version is the most popular one (just compare global charts of Bukkit with global charts of Sponge). If you would just throw all this information together this details would get lost.

**Comment by jamierocks** [comment]
Absolutely, both is what I'd like. I'm rather well acquainted with CanaryLib, I have no issue making a pr for the Canary Metrics class.

**Comment by BtoBastian** [comment]
Can you please tell me the name of the account on bStats for CanaryMod? I'll register a "server implementation" for Canary and transfer it to this account.

**Comment by jamierocks** [comment]
Sure, `jamierocks` for now - assuming that this could be changed in the future (although I doubt this'll be necessary).
I guess Sponge has the same issue that I'll have here - no distinction between different implementations (SpongeForge, SpongeVanilla - NeptuneMod, CanaryMod).

**Comment by BtoBastian** [comment]
Hey,
I transfered https://bstats.org/plugin/server-implementation/CanaryMod to your account.
It's from the type "Server Implementation" which has way less default charts as normal plugins (in order to make it as flexible as possible with all different types of implementations) which means that you have to create custom charts for stuff like players, etc. Just ask me, if you need help with creating the custom charts.
The Metric class and some additional information can be found here: https://github.com/PaperMC/Paper/issues/646#issuecomment-293876337.
As different implementation use other config "styles", you also have to create the config file to disable bStats yourself. I'm not familiar with CanaryMod so this is up to you. I recommend to create a seperate folder for bStats which can also be used by plugins.

**Comment by BtoBastian** [comment]
> I guess Sponge has the same issue that I'll have here - no distinction between different implementations (SpongeForge, SpongeVanilla - NeptuneMod, CanaryMod).

Since version 1.2 of the Metrics classes, they also send the server implementation, too (see https://github.com/BtoBastian/bStats-Metrics/blob/master/bstats-sponge-lite/src/main/java/org/bstats/sponge/MetricsLite.java#L201). This data is not displayed on the website at the moment, but it will after enough plugins switched to the new Metrics class.

**Comment by BtoBastian** [comment]
@jamierocks Is there still interest in this? If you need help with the custom charts, I can create them for you.

**Comment by jamierocks** [comment]
Absolutely, I've just been unbelievable busy recently :(

**Comment by DoNotSpamPls** [comment]
👀
